os/arch/arm/src/imxrt/imxrt_usbhost.c

```diff
@@ -0,0 +1,343 @@
+
+static struct usbhost_connection_s *g_ehciconn;
+
+/* Overcurrent interrupt handler */
+
+#if 0							/* Not yet implemented */
+static xcpt_t g_ochandler;
+#endif
+
+/************************************************************************************
+ * Private Functions
+ ************************************************************************************/
+
+/************************************************************************************
+ * Name: ehci_waiter
+ *
+ * Description:
+ *   Wait for USB devices to be connected to the EHCI root hub.
+ *
+ ************************************************************************************/
+
+static int ehci_waiter(int argc, char *argv[])
+{
+	FAR struct usbhost_hubport_s *hport;
+
+	uinfo("ehci_waiter:  Running\n");
+	for (;;) {
+		/* Wait for the device to change state */
+
+		DEBUGVERIFY(CONN_WAIT(g_ehciconn, &hport));
+		syslog(LOG_INFO, "ehci_waiter: %s\n", hport->connected ? "connected" : "disconnected");
+
+		/* Did we just become connected? */
+
+		if (hport->connected) {
+			/* Yes.. enumerate the newly connected device */
+
+			(void)CONN_ENUMERATE(g_ehciconn, hport);
+		}
+	}
+
+	/* Keep the compiler from complaining */
+
+	return 0;
+}
+
+/************************************************************************************
+ * Public Functions
+ ************************************************************************************/
+
+/************************************************************************************
+ * Name: imxrt_usbhost_bootinitialize
+ *
+ * Description:
+ *   Called from imxrt_boardinitialize very early in inialization to setup USB
+ *   host-related GPIO pins for the LPC-H3131 board.
+ *
+ *   SIGNAL      GPIO
+ *   ----------- -------
+ *   #OTG_PWR_E  GPIO19
+ *   #OTG_OVRCR  GPIO20
+ *
+ ************************************************************************************/
+
+void weak_function imxrt_usbhost_bootinitialize(void)
+{
+	/* Configure output pin to drive VBUS power (initial state: power off) */
+
+	imxrt_config_gpio(GPIO_USB_OTG1_PWR_1);
+
+	/* Configure input pin to detect overrcurrent errors */
+
+	imxrt_config_gpio(GPIO_USB_OTG1_OC_1);
+
+	/* Configure to receive interrupts on the overrcurrent input pin */
+#warning Missing logic
+}
+
+/***********************************************************************************
+ * Name: imxrt_usbhost_initialize
+ *
+ * Description:
+ *   Called at application startup time to initialize the USB host functionality.
+ *   This function will start a thread that will monitor for device
+ *   connection/disconnection events.
+ *
+ ***********************************************************************************/
+
+int imxrt_usbhost_initialize(void)
+{
+	pid_t pid;
+	int ret;
+
+#if 0
+	/* First, register all of the class drivers needed to support the drivers
+	 * that we care about
+	 */
+
+#ifdef CONFIG_USBHOST_HUB
+	/* Initialize USB hub support */
+
+	ret = usbhost_hub_initialize();
```

Comment:
isn't the hub initialization is necessary? Why you have commented it?

Reply:
I removed it